src/app/personside/kontrollsporsmal/SporsmalOgSvarContainer.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import { KontrollSpørsmålState, Spørsmål } from '../../../redux/restReducers/kontrollSporsmal/types';
+import { AppState } from '../../../redux/reducers';
+import SpørsmålOgSvar, { getFeilTekst } from './SporsmalOgSvar';
+import { AsyncDispatch } from '../../../redux/ThunkTypes';
+import { setKontrollSpørsmål } from '../../../redux/restReducers/kontrollSporsmal/actions';
+import { connect } from 'react-redux';
+import { isLoaded, RestReducer } from '../../../redux/restReducers/restReducer';
+import { PersonRespons } from '../../../models/person/person';
+import { KRRKontaktinformasjon } from '../../../models/kontaktinformasjon';
+import { kontaktInformasjonSpørsmål, personInformasjonSpørsmål, SpørsmålsExtractor } from './SporsmalExtractors';
+import { shuffle } from '../../../utils/list-utils';
+import { loggEvent } from '../../../utils/frontendLogger';
+import { getSaksbehandlerIdent } from '../../../utils/loggInfo/getSaksbehandlerIdent';
+
+interface StateProps {
+    personinformasjon: RestReducer<PersonRespons>;
+    kontaktinformasjon: RestReducer<KRRKontaktinformasjon>;
+    kontrollSpørsmål: KontrollSpørsmålState;
+}
+
+interface DispatchProps {
+    setSpørsmål: (spørsmål: Spørsmål[]) => void;
+}
+
+export type Props = StateProps & DispatchProps;
+
+class SpørsmålOgSvarContainer extends React.PureComponent<Props> {
+    componentDidMount() {
+        this.props.setSpørsmål(this.lagSpørsmål());
+        loggEvent('Visning', 'Kontrollsporsmal', undefined, {ident: getSaksbehandlerIdent()});
+    }
+
+    componentDidUpdate(nextProps: Readonly<Props>) {
+        this.oppdaterSpørsmål();
+    }
+
+    render() {
+        if (!this.props.kontrollSpørsmål.spørsmål || this.props.kontrollSpørsmål.spørsmål.length === 0) {
+            return getFeilTekst();
+        }
+
+        return (
+            <SpørsmålOgSvar spørsmål={this.props.kontrollSpørsmål.spørsmål[0]}/>
+        );
+    }
+
+    private oppdaterSpørsmål() {
+        const spørsmål = this.lagSpørsmål();
+        const nyeEllerEndredeSpørsmål = spørsmål.filter(spm => !this.spørsmålEksisterer(spm));
+        if (nyeEllerEndredeSpørsmål.length > 0) {
+            this.props.setSpørsmål(spørsmål);
+        }
+
+    }
+
+    private lagSpørsmål(): Spørsmål[] {
+        let spørsmål: Spørsmål[] = [];
+
+        spørsmål = spørsmål.concat(extractSpørsmål(this.props.personinformasjon, personInformasjonSpørsmål));
```

Comment:
concat er mutating tror jeg, her kan du bruke nananana `...spreadOperator` 🎵

Reply:
Trooor jeg sjekka MDN for at den lager en ny :)